### DISCUSSION TITLE
dagster dask worker can not assess code locations.

### DISCUSSION THREAD

**Discussion by zqbake** [dagster-io/dagster#12296] [Q&A]
1. dask worker can not access code locations, how to solve?
I was using dagster-dask[distributed], and meet error while running jobs. This is because the dask worker and dagster are in different file systems and thus the worker can not access the job. One solution is to use a shared network file system to store codes, If there any other ways? like storing the code location in persistant databse.
2. If there are other types of code locations instead of pyton file/module?
I want to dynamically generate jobs from graph representation files(yaml, json...) or from frontend. So there are no actual job python files or code locations. How can I achieve this? If dagster-dask can execute job without a pyton file/module?

**Comment by OwenKephart** [comment]
For your first question, it's true that you will need to make sure your code is installed on your dask cluster in order for it to function. For that, I'll refer you out to this StackOverflow post, which covers the options fairly well: https://stackoverflow.com/questions/60643162/what-is-the-best-approach-to-make-all-the-imports-with-in-the-service-available. 

For your second question, even if you're dynamically generating your definitions/repository from a file, the code that you use to do that will still be in a python file that you can reference as your code location. I.e. you'll need some file like:

```
from dagster import Definitions

with open("some_file.yaml") as file:
    defs = Definitions(...) # somehow create definitions from the file 
```

This python file (or the module it lives in) can be referenced as your code location.

**Reply by zqbake** [reply]
> For your first question, it's true that you will need to make sure your code is installed on your dask cluster in order for it to function. For that, I'll refer you out to this StackOverflow post, which covers the options fairly well: https://stackoverflow.com/questions/60643162/what-is-the-best-approach-to-make-all-the-imports-with-in-the-service-available.
> 
> For your second question, even if you're dynamically generating your definitions/repository from a file, the code that you use to do that will still be in a python file that you can reference as your code location. I.e. you'll need some file like:
> 
> ```
> from dagster import Definitions
> 
> with open("some_file.yaml") as file:
>     defs = Definitions(...) # somehow create definitions from the file 
> ```
> 
> This python file (or the module it lives in) can be referenced as your code location.

Can I add another CodePointer type like `Storage` Type? Then I can load the definition stored in persistant Postgress database.

**Reply by zqbake** [reply]
> For your first question, it's true that you will need to make sure your code is installed on your dask cluster in order for it to function. For that, I'll refer you out to this StackOverflow post, which covers the options fairly well: https://stackoverflow.com/questions/60643162/what-is-the-best-approach-to-make-all-the-imports-with-in-the-service-available.
> 
> For your second question, even if you're dynamically generating your definitions/repository from a file, the code that you use to do that will still be in a python file that you can reference as your code location. I.e. you'll need some file like:
> 
> ```
> from dagster import Definitions
> 
> with open("some_file.yaml") as file:
>     defs = Definitions(...) # somehow create definitions from the file 
> ```
> 
> This python file (or the module it lives in) can be referenced as your code location.

And also. Do I need to lick `reload definitions` in dagit ui or restart dagster code locations grpc server whenever the code locations has changed(add/remove a new code locations)？. If this mannual process can be achived automatically(through a file wather)？
